src/data/repositories/SurveyFormD2Repository.ts

```diff
@@ -348,30 +369,84 @@ export class SurveyD2Repository implements SurveyRepository {
         }
     }
 
-    getSurveys(programId: Id, orgUnitId: Id): FutureData<Survey[]> {
+    getSurveyNameBySurveyFormType(
+        surveyFormType: SURVEY_FORM_TYPES,
+        options: {
+            eventId: string;
+            surveyName: string;
+            orgUnitName: string | undefined;
+            hospitalCode: string;
+            wardCode: string;
+            patientCode: string;
+        }
+    ): string {
+        switch (surveyFormType) {
+            case "PPSSurveyForm":
+                return options.surveyName !== "" ? options.surveyName : options.eventId;
+            case "PPSCountryQuestionnaire":
+                return options.orgUnitName && options.orgUnitName !== ""
+                    ? options.orgUnitName
+                    : options.eventId;
+            case "PPSHospitalForm":
+                return options.hospitalCode !== "" ? options.hospitalCode : options.eventId;
+            case "PPSWardRegister":
+                return options.wardCode !== "" ? options.wardCode : options.eventId;
+            case "PPSPatientRegister":
+                return options.patientCode !== "" ? options.patientCode : options.eventId;
+            default:
+                return "";
+        }
+    }
+
+    getSurveys(
+        surveyFormType: SURVEY_FORM_TYPES,
+        programId: Id,
+        orgUnitId: Id
+    ): FutureData<Survey[]> {
+        const ouMode =
+            orgUnitId !== "" &&
+            (programId === PPS_WARD_REGISTER_ID ||
+                programId === PPS_HOSPITAL_FORM_ID ||
+                programId === PPS_PATIENT_REGISTER_ID)
+                ? "DESCENDANTS"
+                : undefined;
         return apiToFuture(
             this.api.tracker.events.get({
                 fields: { $all: true },
                 program: programId,
                 orgUnit: orgUnitId,
+                ouMode: ouMode,
             })
         ).flatMap(events => {
-            const surveys: Survey[] = events.instances.map(event => {
-                const startDateString = event.dataValues.find(
-                    dv => dv.dataElement === START_DATE_DATAELEMENT_ID
-                )?.value;
-
-                const surveyType = event.dataValues.find(
-                    dv => dv.dataElement === SURVEY_TYPE_DATAELEMENT_ID
-                )?.value;
-
-                const surveyCompleted = event.dataValues.find(
-                    dv => dv.dataElement === SURVEY_COMPLETED_DATAELEMENT_ID
-                )?.value;
+            const surveys = events.instances.map(event => {
+                let startDateString,
```

Comment:
Applying functional programming from coding-dojo: Use immutable, declarative transformations (avoid let, +=, push, for/while/forEach loops) whenever possible. Maybe instead of using let and a forEach, it's better to use multiple find functions and create const variables

Reply:
@anagperal Nice catch! We discussed this same code in our previous coding dojo sessionwith Arnau and after discussion he has provided a solution of using object/hashmap to store the data element to property mapping. Will implement this in my next PR and tag you.

Reply:
yes please, I need to learn that solution!! :pray:

Reply:
done in https://github.com/EyeSeeTea/amr-surveys/pull/11